src/App.tsx

```diff
@@ -1,8 +1,64 @@
-import React from 'react';
+import React, { useState, FC } from 'react';
 import './App.css';
+import { TodosList } from './components/TodosList/TodosList';
+import { getUsers, getTodos } from './api';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App: FC = () => {
+  const [isLoading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<TodoWithUser[]>([]);
+
+  const loadTodos = async () => {
+    setLoading(true);
+    const [todosLoaded, users] = await Promise.all([getTodos(), getUsers()]);
+
+    const todosWithUsers = todosLoaded.map(todo => ({
+      ...todo,
+      user: users.find(user => user.id === todo.userId),
+    }));
+
+    setTodos(todosWithUsers);
+  };
+
+  const sortByTitle = () => {
+    setTodos([...todos].sort((a, b) => a.title.localeCompare(b.title)));
+  };
+
+  const sortByName = () => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    setTodos([...todos].sort((a, b) => a.user!.name.localeCompare(b.user!.name)));
```

Comment:
давай стараться не отключать линтер
```suggestion
    setTodos([...todos].sort((a, b) => {
      if (a.user && b.user) {
        return a.user.name.localeCompare(b.user.name);
      }

      return 0;
    }));
```

Reply:
sure, fixing

Reply:
done